### PR TITLE
[docs] Adding Dataset usage examples

### PR DESCRIPTION
**Pull request by gilad-shaham**
Added some documentation of dataset usage
Also some minor fixes to the documentation

### CODE REVIEW

File: docs/data-management-and-versioning.md [hunk 1]
```diff
@@ -37,11 +38,56 @@ Where `key` is the the name of the artifact and `df` is the DataFrame. By defaul
...
+source_url = 'http://iguazio-sample-data.s3.amazonaws.com/iris_dataset.csv'
+# Run get-data function locally
+get_data_run = run_local(name='get_data',
+                         handler=get_data,
+                         inputs={'source_url': source_url},
+                         project=project_name,
+                         artifact_path=artifact_path)
+```
+
+The dataset location is returned in the `output` field, therefore you can get the location by calling `get_data_run.outputs['iris_dataset']` and use the `get_dataitem` function to get the dataset itself.
```
Comment (Hedingber) [inline docs/data-management-and-versioning.md:76]:
```suggestion
The dataset location is returned in the `outputs` field, therefore you can get the location by calling `get_data_run.outputs['iris_dataset']` and use the `get_dataitem` function to get the dataset itself.

```

File: docs/data-management-and-versioning.md [hunk 2]
```diff
@@ -161,8 +207,21 @@ Storing plots is useful to visualize the data and to show any information regard
 For example, the following code creates a confusion matrix plot using [sklearn.metrics.plot_confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.plot_confusion_matrix.html#sklearn.metrics.plot_confusion_matrix) and stores the plot in the artifact repository:
 
 ``` python
-    cmd = metrics.plot_confusion_matrix(model, xtest, ytest, normalize='all', values_format='.2g', cmap=plt.cm.Blues)
-    context.log_artifact(PlotArtifact('confusion-matrix', body=cmd.figure_), local_path='plots/confusion_matrix.html')
+from mlrun.artifacts import PlotArtifact
+from mlrun.mlutils import gcf_clear
+
+gcf_clear(plt)
+cmd = metrics.plot_confusion_matrix(model, xtest, ytest, normalize='all', values_format = '.2g', cmap=plt.cm.Blues)
```
Comment (Hedingber) [inline docs/data-management-and-versioning.md:214]: what does cmd means ?
The only thing I can think of is shortcut of command which I don't understand how it's related to the context
we can probably find a better naming

File: docs/job-submission-and-tracking.md [hunk 3]
```diff
@@ -20,11 +20,11 @@
   - [Using the MLRun CLI to Run an MLRun Service](#using-the-mlrun-cli-to-run-an-mlrun-service)
 
 ##  Experiment Tracking
-Experiment tracking enables you to store every action and result in your project. It is a convenient way to go back to previous results and compare different artifacts. You will find 3 main sections within the your project:
-1. [**Artifacts**](#artifact): Any data stored is considered an artifact. Artifacts are versioned and enable you to compare different outputs of the executed Jobs
-2. [**Functions**](#functions): The code in your project is stored in functions that are versioned. Functions can the functions you wrote, or externally loaded functions, such as functions that originate from the [MLRun Functions Marketplace](https://github.com/mlrun/functions)
-3. [**Jobs**](#jobs): Allows you to review anything you executed, and review the execution outcome
-4. [**Pipelines**](#pipelines): Reusable end-to-end ML workflows
+Experiment tracking enables you to store every action and result in your project. It is a convenient way to go back to previous results and compare different artifacts. You will find the following sections within the your project:
```
Comment (Hedingber) [inline docs/job-submission-and-tracking.md:23]:
```suggestion
Experiment tracking enables you to store every action and result in your project. It is a convenient way to go back to previous results and compare different artifacts. You will find the following sections within your project:
```

**Review by Hedingber — CHANGES_REQUESTED** [review]
Looks good! 
Thanks for putting effort on improving the docs
Some minor comments

File: docs/data-management-and-versioning.md (hunk 2 above)
Comment (gilad-shaham) [reply docs/data-management-and-versioning.md:215]: I took it from mlutils.eval_class_mode. I'll change the name here so it's clearer (likely the original meaning is confusion matrix diagram)

File: docs/data-management-and-versioning.md [hunk 4]
```diff
@@ -37,11 +38,56 @@ Where `key` is the the name of the artifact and `df` is the DataFrame. By defaul
...
+We can run this function locally or as a job. For example if we run it locally:
+
+``` python
+source_url = 'http://iguazio-sample-data.s3.amazonaws.com/iris_dataset.csv'
+# Run get-data function locally
+get_data_run = run_local(name='get_data',
+                         handler=get_data,
+                         inputs={'source_url': source_url},
+                         project=project_name,
+                         artifact_path=artifact_path)
```
[the remark this answers is not in the document]
Comment (gilad-shaham) [reply docs/data-management-and-versioning.md:80]: Done

**Pull request merged**